clients/callback_client.go

```diff
@@ -157,3 +176,52 @@ func (ts TranscodeStatus) String() string {
 	}
 	return "unknown"
 }
+
+// The various status messages we can send
+
+type TranscodeStatusMessage struct {
+	CompletionRatio float64 `json:"completion_ratio"` // No omitempty or we lose this for 0% completion case
+	Error           string  `json:"error,omitempty"`
+	Retriable       *bool   `json:"retriable,omitempty"` // Has to be a pointer or we can't differentiate omission from 'false'
+	Status          string  `json:"status,omitempty"`
+	Timestamp       int64   `json:"timestamp"`
+}
+
+type InputVideoTrack struct {
+	Fps         int     `json:"fps,omitempty"`
+	Type        string  `json:"type,omitempty"`
+	Codec       string  `json:"codec,omitempty"`
+	Width       int     `json:"width,omitempty"`
+	Height      int     `json:"height,omitempty"`
+	Bitrate     int     `json:"bitrate,omitempty"`
+	Duration    float64 `json:"duration,omitempty"`
+	PixelFormat string  `json:"pixel_format,omitempty"`
+	Channels    int     `json:"channels,omitempty"`
+	SampleRate  int     `json:"sampleRate,omitempty"`
+	SizeBytes   int     `json:"size,omitempty"`
```

Comment:
Might be worth reordering/categorizing the fields here just so it's easier to understand whats higher-level, what's specific for given codecs, etc. e.g.
https://github.com/livepeer/go-api-client/blob/76a0884161c19aea4c15f99897540b3880df5aad/api.go#L317-L332

Also, I like to remove the `omitempty` on fields that are not optional, like `type` (and maybe `codec`). Not necessarily following any guideline tho I guess.

Reply:
Split it out into seperate sub-structs now, let me know what you think.

Removed the omitemptys - we can add back in as needed in the future

Reply:
Ah neat! I liked the separate structs.

Could add the `omitempty` at least on the audio/video track fields, since those are not present for the other track type.

Reply:
Good point, done